src/test/java/com/openlattice/rehearsal/organization/AssemblerTest.kt

```diff
@@ -626,6 +665,150 @@ class AssemblerTest : MultipleAuthenticatedUsersBase() {
+                .contains(OrganizationEntitySetFlag.DATA_UNSYNCHRONIZED))
+        val testData = TestDataFactory.randomStringEntityData(numberOfEntities, et.properties).values.toList()
+        val ids = dataApi.createEntities(es.id, testData)
+        val testDataWithIds = ids.zip(testData).toMap()
+
+        Assert.assertTrue(organizationsApi.getOrganizationEntitySets(organizationID)[es.id]!!
+                .contains(OrganizationEntitySetFlag.DATA_UNSYNCHRONIZED))
+
+        // data is not supposed to be there, only the columns until automatic refresh
+        organizationDataSource.connection.use { connection ->
+            connection.createStatement().use { stmt ->
+                val rs = stmt.executeQuery(TestAssemblerConnectionManager.selectFromEntitySetSql(es))
+                // all columns are there
+                (1..rs.metaData.columnCount).forEach {
+                    val columnName = rs.metaData.getColumnName(it)
+                    if (columnName != PostgresColumn.ID.name && columnName != PostgresColumn.ENTITY_SET_ID.name
+                            && columnName != ResultSetAdapters
+                                    .mapMetadataOptionToPostgresColumn(MetadataOption.ENTITY_KEY_IDS)) {
+                        Assert.assertTrue(propertyFqns.values.contains(columnName))
+                    }
+                }
+                // no data is there yet
+                Assert.assertFalse(rs.next())
+            }
+        }
+
+        // wait until automatic refresh
+        Thread.sleep(refreshRate.toLong() * 60 * 1000)
+
+
+        // check if data is in org database
+        organizationDataSource.connection.use { connection ->
+            connection.createStatement().use { stmt ->
+                val rs = stmt.executeQuery(TestAssemblerConnectionManager.selectFromEntitySetSql(es))
+
+                var index = 0
+                Assert.assertTrue(rs.next())
+                Assert.assertEquals(es.id, ResultSetAdapters.entitySetId(rs))
+                do {
+                    val id = ResultSetAdapters.id(rs)
+                    Assert.assertTrue(ids.contains(id))
+                    propertyFqns.forEach { propertyId, fqn ->
+                        Assert.assertEquals(
+                                testDataWithIds.getValue(id).getValue(propertyId).first(),
+                                getStringResult(rs, fqn))
+                    }
+                    index++
+
+                } while (rs.next())
+
+                Assert.assertEquals(numberOfEntities, index)
+            }
+        }
+
+
+        // set refresh rate to 4 mins
+        refreshRate = 4
+        organizationsApi.updateRefreshRate(organizationID, es.id, refreshRate)
+
+        // delete all data
+        dataApi.deleteAllEntitiesFromEntitySet(es.id, DeleteType.Hard)
+
+        // wait until automatic refresh
+        Thread.sleep(refreshRate.toLong() * 60 * 1000)
```

Comment:
might be worth shortening the refresh rate because of all the `Thread.sleep`s

Reply:
:+1: